src/esnext.coffee

```diff
@@ -0,0 +1,157 @@
+###
+Cache for source code transpiled by 6to5.
+
+Inspired by https://github.com/atom/atom/blob/6b963a562f8d495fbebe6abdbafbc7caf705f2c3/src/coffee-cache.coffee.
+###
+
+crypto = require 'crypto'
+fs = require 'fs-plus'
+path = require 'path';
+to5 = require '6to5-core'
+
+stats =
+  hits: 0
+  misses: 0
+
+defaultOptions =
+  # The Chrome dev tools will show the original version of the file
+  # when the source map is inlined.
+  'sourceMap': 'inline'
+
+  # Because Atom is currently packaged with a fork of React v0.11,
+  # it makes sense to use the --react-compat option so the React
+  # JSX transformer produces pre-v0.12 code.
+  'reactCompat': true
+
+  # Blacklisted features do not get transpiled. Features that are
+  # natively supported in the target environment should be listed
+  # here. Because Atom uses a bleeding edge version of Node/io.js,
+  # I think this can include es6.arrowFunctions, es6.classes, and
+  # possibly others, but I want to be conservative.
+  'blacklist': [
+    'useStrict'
+  ]
+
+  # Includes support for es7 features listed at:
+  # http://6to5.org/docs/usage/transformers/#es7-experimental-.
+  'experimental': true,
+
+  'optional': [
+    # Target a version of the regenerator runtime that
+    # supports yield so the transpiled code is cleaner/smaller.
+    'asyncToGenerator'
+  ]
+
+###
+shasum - Hash with an update() method.
+value - Must be a value that could be returned by JSON.parse().
+###
+updateDigestForJsonValue = (shasum, value) ->
+  # Implmentation is similar to that of pretty-printing a JSON object, except:
+  # * Strings are not escaped.
+  # * No effort is made to avoid trailing commas.
+  # These shortcuts should not affect the correctness of this function.
+  if typeof value is 'string'
+    shasum.update('"', 'utf8')
+    shasum.update(value, 'utf8')
+    shasum.update('"', 'utf8')
+  else if typeof value is 'boolean' or typeof value is 'number'
```

Comment:
This could be the shorter `else if typeof value in ['boolean', 'number']` and CoffeeScript will compile to the same thing as this JS-wise.